multicluster/controllers/multicluster/member/clusterset_controller.go

```diff
@@ -130,20 +159,63 @@ func (r *MemberClusterSetReconciler) Reconcile(ctx context.Context, req ctrl.Req
 			err = r.Update(context.TODO(), clusterSet)
 			if err != nil {
 				klog.ErrorS(err, "Failed to update ClusterSet's ClusterID", "clusterset", req.NamespacedName)
-				return ctrl.Result{}, err
+				return err
 			}
 		}
+
+		r.clusterSetConfig = clusterSet.DeepCopy()
+
+		err = r.createRemoteCommonArea(clusterSet)
+		if err != nil {
+			return err
+		}
+		return nil
 	}
-	r.clusterSetConfig = clusterSet.DeepCopy()
 
-	err = r.createOrUpdateRemoteCommonArea(clusterSet)
-	if err != nil {
+	if err := processClusterSet(); err != nil {
 		return ctrl.Result{}, err
 	}
 
+	if clusterSetCreated {
+		// The CommonArea creation succeeded here and so notify StaleController to
+		// clean up stale imported resources and ResourceExports.
+		r.commonAreaCreationCh <- struct{}{}
+	}
 	return ctrl.Result{}, nil
 }
 
+func (r *MemberClusterSetReconciler) cleanUpResources(ctx context.Context, secretChangeOnly bool) error {
+	if r.remoteCommonArea != nil {
+		if !secretChangeOnly {
```

Comment:
Your current approach should also work, but I feel it is not easy to understand that `cleanUpResources` takes this `secretChangeOnly` argument and only stops commonArea without really doing any cleanup if true.

So, would you consider the approach I suggested - call  `cleanUpResources` only when cleanup is needed, that is when `r.clusterID != newClusterID`, and in `createRemoteCommonArea` stops the current commonArea if it is not nil? Yes, it means there can be two places to stop commonArea, but I feel that is still easier to understand than checking `secretChangeOnly `in `secretChangeOnly`

Reply:
Thanks for explanation, I have refined this part, please help to check.